### PR TITLE
Fixes #17268: rudder agent run returns exit code 1 when everything is ok

### PR DESCRIPTION
**Pull request by amousset**

### CODE REVIEW

File: share/lib/reports.awk [hunk 1]
```diff
@@ -465,7 +465,7 @@ END {
   printf "%s%-80.80s%s\n", white, padding, normal;
 
   # Set return code
-  if (run_error+audit_error+audit_noncompliant+enforce_error != 0) {
+  if (run_error+audit_error+enforce_error != 0) {
```
Comment (peckpeck) [inline share/lib/reports.awk:468]: i validated, but this means that we want non compliant to be ok ?

Comment (fanf) [reply share/lib/reports.awk:468]: I'm not sur I understand either. Actually, I'm not sure of the semantic of the return code, but `non compliance` is as much an error as `impossible to enforce`, I think. If my ssh key does't match the expected set of authorized one, it's as important as if in enforce, it was that and I wasn't able to change it. Or is this audit error? And if so, what is `noncompliant` ?

Comment (amousset) [reply share/lib/reports.awk:468]: It means we consider non-compliant to be normal and expected agent behavior. Errors or aborted runs are not normal and should be fixed by the user.

Comment (amousset) [reply share/lib/reports.awk:468]: In fact I think error reports should not change the exit code to 1. Aborted agents are the only real problems the agent itself can have.

Comment (fanf) [reply share/lib/reports.awk:468]: I agree with that, but if so: why audit error and enforce error change it to 1?

Comment (fanf) [reply share/lib/reports.awk:468]: (ie, we diferentiate from "expected" errors that are just part of the business domain, and "unexpect" defaicts that should be processed outside of business domain. An enforce in error is expected, an agent run crash is not)

Comment (amousset) [reply share/lib/reports.awk:468]: They shouldn't :), il make another PR.

Comment (amousset) [reply share/lib/reports.awk:468]: https://github.com/Normation/rudder-agent/pull/299 and https://github.com/Normation/rudder-agent/pull/298 for the doc

**Pull request merged**